backend/asmlink.ml

```diff
@@ -37,21 +37,21 @@ exception Error of error
 
 (* Consistency check between interfaces and implementations *)
 
-module Cmi_consistbl = Consistbl.Make (Misc.Stdlib.String)
+module Cmi_consistbl = Consistbl.Make (String)
 let crc_interfaces = Cmi_consistbl.create ()
-let interfaces = ref ([] : string list)
+let interfaces = ref String.Set.empty
```

Comment:
Have you compared the performance of using a `String.Set.t ref` with `unit String.Tbl.t` for this?

Reply:
I was just thinking the same this morning, I will try that.

Reply:
I wrote the following simple benchmark to check the performance of `add` and `fold` on these two data structures.  The unit hash table version took 11 seconds and the set-in-ref versions took 52 seconds, so I think the answer is clear...

```
module String_set = Set.Make (struct
  type t = string
  let compare = String.compare
end)

module String_tbl = Hashtbl.Make (struct
  type t = string
  let equal = String.equal
  let hash = Hashtbl.hash
end)

let num_strings = 10_000_000

let () = Random.self_init ()

let strings =
  List.init num_strings (fun i ->
    let length = 8 + Random.int 10 in
    String.init length (fun _i ->
      Char.chr (65 + Random.int 50)) ^ ".cmi")

let (_ : Unix.process_status) = Unix.system "date"

let tbl = String_tbl.create 100

let () =
  List.iter (fun s -> String_tbl.add tbl s ()) strings

let () =
  String_tbl.fold (fun _s () acc -> acc) tbl ()

let (_ : Unix.process_status) = Unix.system "date"

let set = ref String_set.empty

let () =
  List.iter (fun s -> set := String_set.add s !set) strings

let () =
  String_set.fold (fun _s acc -> acc) !set ()

let (_ : Unix.process_status) = Unix.system "date"
```